deepchecks/nlp/context.py

```diff
@@ -326,14 +329,45 @@ def __init__(
             test_dataset = TextData.cast_to_dataset(test_dataset)
             if test_dataset.name is None:
                 test_dataset.name = 'Test'
+
         # If both dataset, validate they fit each other
         if train_dataset and test_dataset:
-            if test_dataset.has_label() and train_dataset.has_label() and not \
-                    train_dataset.validate_textdata_compatibility(test_dataset):
+            if (
+                test_dataset.has_label()
+                and train_dataset.has_label()
```

Comment:
why is this needed? validate_textdata_compatibility should be able to work regardless is labels are present or not

Reply:
hm, I do not know/ I did not add this, I just moved each part of the condition exp on the new line

Reply:
ok yea i see